src/transform/FooterReadMore.js

```diff
@@ -0,0 +1,336 @@
+  gsrinfo: '',
+  gsrlimit: count,
+  gsrnamespace: 0,
+  gsroffset: 0,
+  gsrprop: 'redirecttitle',
+  gsrsearch: `morelike:${title}`,
+  gsrwhat: 'text',
+  ns: 'ppprop',
+  pilimit: count,
+  piprop: 'thumbnail',
+  pithumbsize: 120,
+  prop: 'pageterms|pageimages|pageprops|revisions|extracts',
+  rrvlimit: 1,
+  rvprop: 'ids',
+  wbptterms: 'description',
+  formatversion: 2
+})
+
+/**
+ * Converts query parameter object to string.
+ * @param {!object} parameters
+ * @return {!string}
+ */
+const stringFromQueryParameters = parameters => Object.keys(parameters)
+  .map(key => `${encodeURIComponent(key)}=${encodeURIComponent(parameters[key])}`)
+  .join('&')
+
+/**
+ * URL for retrieving 'Read more' pages for a given title.
+ * Leave 'baseURL' null if you don't need to deal with proxying.
+ * @param {!string} title
+ * @param {!number} count Number of `Read more` items to fetch for this title
+ * @param {?string} baseURL
+ * @return {!sring}
+ */
+const readMoreQueryURL = (title, count, baseURL) =>
+  `${baseURL || ''}/w/api.php?${stringFromQueryParameters(queryParameters(title, count))}`
+
+/**
+ * Fetch error handler.
+ * @param {!string} statusText
+ * @return {void}
+ */
+const fetchErrorHandler = statusText => {
+  // TODO: figure out if we want to hide the 'Read more' header in cases when fetch fails.
+  // console.log(`statusText = ${statusText}`);
+}
+
+/**
+ * Fetches 'Read more' pages.
+ * @param {!string} title
+ * @param {!number} count
+ * @param {!string} containerID
+ * @param {?string} baseURL
+ * @param {ShownReadMorePagesHandler} showReadMorePagesHandler
+ * @param {SaveButtonClickHandler} saveButtonClickHandler
+ * @param {TitlesShownHandler} titlesShownHandler
+ * @param {!Document} document
+ * @return {void}
+ */
+const fetchReadMore = (title, count, containerID, baseURL, showReadMorePagesHandler,
+  saveButtonClickHandler, titlesShownHandler, document) => {
+  const xhr = new XMLHttpRequest() // eslint-disable-line no-undef
+  xhr.open('GET', readMoreQueryURL(title, count, baseURL), true)
+  xhr.onload = () => {
+    if (xhr.readyState === XMLHttpRequest.DONE) { // eslint-disable-line no-undef
+      if (xhr.status === 200) {
+        showReadMorePagesHandler(
+          JSON.parse(xhr.responseText).query.pages,
+          containerID,
+          saveButtonClickHandler,
+          titlesShownHandler,
+          document
+        )
+      } else {
+        fetchErrorHandler(xhr.statusText)
+      }
+    }
+  }
+  xhr.onerror = e => {
+    fetchErrorHandler(xhr.statusText)
+  }
+  xhr.send()
+}
+
+/**
+ * Updates save button bookmark icon for saved state.
+ * @param {!HTMLDivElement} button
+ * @param {!boolean} isSaved
+ * @return {void}
+ */
+const updateSaveButtonBookmarkIcon = (button, isSaved) => {
+  const unfilledClass = 'pagelib_footer_readmore_bookmark_unfilled'
+  const filledClass = 'pagelib_footer_readmore_bookmark_filled'
+  button.classList.remove(unfilledClass)
+  button.classList.remove(filledClass)
```

Comment:
Since add won't doubly add, consider reducing this to: `button.classList.remove(isSaved ? unfilledClass : filledClass)`.

https://developer.mozilla.org/en-US/docs/Web/API/Element/classList